client/domain/PreRequisiteEvalResultObjectFactory.js

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of
+ * PreRequisiteCheckResult domain objects.
+ */
+
+tie.factory('PreRequisiteEvalResultObjectFactory', [
+  function() {
+    var PreRequisiteEvalResult = function(
+      code, preReqErrorMessage, starterCode) {
+      this._code = code;
+      this._preReqErrorMessage = preReqErrorMessage;
+      this._starterCode = starterCode;
+    };
+
+    // Instance methods
+    PreRequisiteEvalResult.prototype.getCode = function() {
```

Comment:
Hi Sean,

I think that depends on when the CodeEvalResult object is created.  If the
CodeEvalResult object is created as a result of a syntax error during the
the initial run of the raw code, then the CodeEvalResult object will
contain the raw code.  If the student submission passes the syntax error
check and proceeds to pre-processing, then the CodeEvalResult object will
contain the pre-processed code including the testing code.

In the PreRequisites branch as merged, PreRequisiteEvalResultObjectFactory
was renamed CodePrereqCheckResultObjectFactory, and the
CodePrereqCheckResult does not store the submitted code (raw or
otherwise).  At present, I believe the stored code in the CodeEvalResult is
only used in the FeedbackGeneratorService's _getBuggyOutputTestFeedback
function, to check that the student has submitted different code before
incrementing the hintIndex for the same error.